packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -502,7 +504,18 @@ export class ImportResolver {
 
         const entries = this.readdirEntriesCached(splitPath[0]);
         const entry = entries.find((entry) => entry.name === splitPath[1]);
-        return entry !== undefined && entry.isFile();
+        if (entry?.isFile()) {
+            return true;
+        }
+
+        if (entry?.isSymbolicLink()) {
+            const realPath = this.fileSystem.realpathSync(path);
```

Comment:
I am sort of wondering if realpath+exists+stat is slower than just doing a single stat to get the missing info, but I suppose isFile does the throw handling and I doubt this is a bottleneck.

Reply:
I don't think this is on any perf-critical paths, especially with all of the caching we have in place now.